ark/analysis/spatial_analysis.py

```diff
@@ -169,71 +173,73 @@ def calculate_channel_spatial_enrichment(dist_matrices_dict, marker_thresholds,
-            dist_mat=dist_matrix, dist_lim=dist_lim, analysis_type="channel",
-            current_fov_data=current_fov_data, current_fov_channel_data=current_fov_channel_data,
-            thresh_vec=thresh_vec, cell_label_col=cell_label_col)
+    if context_col is not None:
+        close_num_rand = np.zeros((*close_num.shape, bootstrap_num), dtype=np.uint16)
 
-        if context_col is not None:
-            close_num_rand = np.zeros((*close_num.shape, bootstrap_num), dtype=np.uint16)
+        context_nums_per_id = \
+            current_fov_data.groupby(context_col)[cell_label_col].apply(list).to_dict()
 
-            context_nums_per_id = \
-                current_fov_data.groupby(context_col)[cell_label_col].apply(list).to_dict()
+        for name_i, name_j in context_pairings:
+            # some FoVs may not have cells with a certain context, so they are skipped here
+            try:
+                context_cell_labels = context_nums_per_id[name_i]
+                context_cell_labels.extend(context_nums_per_id[name_j])
+            except KeyError:
+                continue
+            context_cell_labels = np.unique(context_cell_labels)
 
-            for name_i, name_j in context_pairings:
-                # some FoVs may not have cells with a certain context, so they are skipped here
-                try:
-                    context_cell_labels = context_nums_per_id[name_i]
-                    context_cell_labels.extend(context_nums_per_id[name_j])
-                except KeyError:
-                    continue
-                context_cell_labels = np.unique(context_cell_labels)
+            context_dist_mat = dist_matrix.loc[context_cell_labels, context_cell_labels]
 
-                context_dist_mat = dist_matrix.loc[context_cell_labels, context_cell_labels]
+            context_pos_labels = [
+                np.intersect1d(mark_pos_label, context_cell_labels)
+                for mark_pos_label in mark_pos_labels
+            ]
 
-                context_pos_labels = [
-                    np.intersect1d(mark_pos_label, context_cell_labels)
-                    for mark_pos_label in mark_pos_labels
-                ]
+            context_mark_nums = [len(cpl) for cpl in context_pos_labels]
 
-                context_mark_nums = [len(cpl) for cpl in context_pos_labels]
+            close_num_rand = close_num_rand + \
+                spatial_analysis_utils.compute_close_cell_num_random(
+                    context_mark_nums, context_pos_labels, context_dist_mat, dist_lim,
+                    bootstrap_num
+                )
 
-                close_num_rand = close_num_rand + \
-                    spatial_analysis_utils.compute_close_cell_num_random(
-                        context_mark_nums, context_pos_labels, context_dist_mat, dist_lim,
-                        bootstrap_num
-                    )
+    close_num_rand = spatial_analysis_utils.compute_close_cell_num_random(
```

Comment:
I think we're missing an `else` before this line - if `context_col is not None`, `compute_close_cell_num_random` is already called and shouldn't be called again. The corresponding `calculate_cluster_spatial_enrichment` function does have an `else`.

Reply:
Good catch, not sure why this never got added in earlier but I'll fix this up.